Removed call to stdio.h

mdouglas90 opened the pull request:
I wrote three functions to replace the call to stdio.h with snprintf. This should solve Issue #126. They are:

data_logging_filename_append_int,
data_logging_filename_append_extension,
is_string_terminated

The is_string_terminated was created to continue doing the error checking that the snprintf allowed. I wasn't able to test the changes made to the previous code but the functions themselves seem to be working. A possible issue that I don't know if you care about is that if the file names are longer than allocated, the functions may try to write to data that wasn't created using malloc(). This can be fixed with setting data_logging->buffer_name_size properly, by adding an additional length parameter to the two functions, or by taking proper action if is_string_terminated fails.

jlecoeur commented:
> A possible issue that I don't know if you care about is that if the file names are longer than allocated, the functions may try to write to data that wasn't created using malloc().

Good point. We can add a length parameter to the functions and when the end of allocated memory is reached, either stop writing, either realloc memory.

mdouglas90 commented:
I added the length parameter and removed the is_string_terminated function. It won't write past the allotted string length and won't add the numbers/.txt if it can't fit. I didn't have it realloc the memory because the pointer was malloced outside of the function and it won't always keep the memory location in the same place.
